bambi/interpret/effects.py

```diff
@@ -1,890 +1,857 @@
-    'slopes' estimates
-
-    Used in 'PredictiveDifferences' to store typed data for the summary dataframe.
+
+def get_defaults(var: str, data: DataFrame) -> Series:
+    """Compute a default value for the given variable in the data.
 
     Parameters
     ----------
-    mean : dict of str to xr.DataArray
-        The mean of the posterior distribution (chains and draws).
-    bounds : dict of str to xr.Dataset
-        The uncertainty interval of the posterior distribution (chains and draws).
-    use_hdi : bool
-        Whether to use the highest density interval (HDI) (True) or quantiles (False).
+    var : str
+        Name of the variable to compute defaults for.
+    data : DataFrame
+        DataFrame containing the variable.
+
+    Returns
+    -------
+    Series
+        A pandas Series containing the default value(s) for the variable.
+        For categorical and integer types, returns the mode.
+        For float types, returns the mean.
+
+    Raises
+    ------
+    TypeError
+        If the variable has an unsupported data type.
     """
+    series = data[var]
 
-    mean: dict[str, xr.DataArray]
-    bounds: dict[str, xr.Dataset]
-    use_hdi: bool
-    bounds_list: list = field(init=False)
-    lower: xr.DataArray = field(init=False)
-    higher: xr.DataArray = field(init=False)
-
-    def __post_init__(self):
-        """
-        Parses the mean and bounds dictionaries into arrays for inserting
-        the 'mean', 'lower', and 'upper' columns into the summary dataframe.
-        """
-        self.bounds_list = [self.bounds[key] for key in self.bounds]
-        self.mean = np.array(list(self.mean.values())).flatten()
-
-        if self.use_hdi:
-            data_var = list(self.bounds_list[0].data_vars)[0]
-            self.lower = np.array(
-                [self.bounds[key][data_var].sel(hdi="lower") for key in self.bounds]
-            ).flatten()
-            self.higher = np.array(
-                [self.bounds[key][data_var].sel(hdi="higher") for key in self.bounds]
-            ).flatten()
-        else:
-            lower = self.bounds_list[0].coords["quantile"].values[0]
-            higher = self.bounds_list[0].coords["quantile"].values[1]
-            self.lower = np.array(
-                [self.bounds[key].sel(quantile=lower) for key in self.bounds]
-            ).flatten()
-            self.higher = np.array(
-                [self.bounds[key].sel(quantile=higher) for key in self.bounds]
-            ).flatten()
-
-
-# pylint: disable=consider-iterating-dictionary
-# pylint: disable=too-many-instance-attributes
-@dataclass
-class PredictiveDifferences:
-    """Computes predictive differences and their uncertainty intervals for
-    'comparisons' and 'slopes' effects and returns a summary dataframe of the results
+    match series.dtype:
+        case pd.CategoricalDtype():
+            # Takes the first mode if there are multiple modes
+            return pd.Series(series.mode().iloc[0], name=var)
```

Comment:
I remember being fooled by `pd.Series.mode` in the past, when we were computing modes by group and the group had either zero rows or all nulls... Just raising some validation may be needed somewhere (perhaps outside of this, I don't know).

```python
import pandas as pd
x = pd.Series([None, None], dtype=pd.CategoricalDtype())
x.mode().iloc[0] # Errors

x = pd.Series([], dtype=pd.CategoricalDtype())
x.mode().iloc[0] # Errors
```

The error occurs because of iloc, not because of mode. That is what's surprising.